girder/api/v1/item.py

```diff
@@ -319,3 +321,48 @@ def copyItem(self, item, params):
         .errorResponse('ID was invalid.')
         .errorResponse('Read access was denied on the original item.', 403)
         .errorResponse('Write access was denied on the parent folder.', 403))
+
+    @access.user
+    def mergeItems(self, params):
+        """
+        Merge multiple items into a single item. The params are passed as
+        JSON in the request body.
+
+        :param folderId: The _id of the parent folder for the new item.
+        :type folderId: str
+        :param name: The name of the item to create.
+        :param description: Item description.
+        """
+        user = self.getCurrentUser()
+        doc = self.getBodyJson()
+        ids = doc['ids']
+        firstItem = self.model('item').load(id=ids[0], user=user)
+        if firstItem is None:
+            raise RestException('One or more of the items to merge was '
+                                'not found.')
+        name = doc.get('name', firstItem['name'])
+        folderId = doc.get('folderId', firstItem['folderId'])
+        folder = self.model('folder').load(id=folderId, user=user,
+                                           level=AccessType.WRITE, exc=True)
```

Comment:
Since we already have the same check in the model layer that sets the default name if name is `None`, could we just use that instead of having it in both places?